src/reactions/MessageReactionList.js

```diff
@@ -0,0 +1,116 @@
+/* @flow strict-local */
+import React, { PureComponent } from 'react';
+import { View } from 'react-native';
+import { TabNavigator, TabBarTop } from 'react-navigation';
+import type { NavigationScreenProp } from 'react-navigation';
+
+import ReactionUserList from './ReactionUserList';
+import { connect } from '../react-redux';
+import type { AggregatedReaction, Dispatch, Message, UserOrBot } from '../types';
+import { Screen, Label, RawLabel } from '../common';
+import { getOwnUser } from '../selectors';
+import aggregateReactions from './aggregateReactions';
+import styles from '../styles';
+import { getAllUsersById } from '../users/userSelectors';
+import tabsOptions from '../styles/tabs';
+import Emoji from '../emoji/Emoji';
+import { objectFromEntries } from '../jsBackport';
+
+const getReactionsTabs = (
+  aggregatedReactions: $ReadOnlyArray<AggregatedReaction>,
+  reactionName: string,
+  usersById: Map<number, UserOrBot>,
+) => {
+  const reactionsTabs = objectFromEntries(
+    aggregatedReactions.map(aggregatedReaction => [
+      aggregatedReaction.name,
+      {
+        screen: () => (
+          <ReactionUserList reactedUserIds={aggregatedReaction.users} users={usersById} />
+        ),
+        navigationOptions: {
+          tabBarLabel: () => (
+            <View style={styles.row}>
+              <Emoji name={aggregatedReaction.name} />
+              <RawLabel style={styles.paddingLeft} text={`${aggregatedReaction.count}`} />
+            </View>
+          ),
+        },
+      },
+    ]),
+  );
+
+  // there can be cases where reaction have been removed by the time user see this screen
+  // or on re-render
+  // so confirm reaction `reactionName` still exists
+  const initialRouteName = aggregatedReactions.find(reaction => reaction.name === reactionName)
+    ? reactionName
+    : '';
```

Comment:
So, the reason I said this would be cleaner after `reactionsTabs` had been constructed is because the code can actually be different and simpler. :slightly_smiling_face: It can use `reactionsTabs`. That's also more directly connected to the point of why this logic is here.

Reply:
oh ya!

Updated 👍